code/game/objects/items/weapons/extinguisher.dm

```diff
@@ -51,7 +52,7 @@
 		// If we're in help intent, don't bash anyone with the
 		// extinguisher
 		user.visible_message("[user] targets [M] with \the [src]", "<span class='info'>You target [M] with \the [src].</span>")
-		return 1
+		return 0

```

Comment:
Returning 1 means not calling afterattack, which means the extinguisher doesn't fire any water.